### ISSUE TITLE
Session.ErrorEventArgs.ErrorCode is meaningless, and this has bad consequences.

### ISSUE THREAD

**Issue by fidergo-stephane-gourichon** [opentok/opentok-windows-sdk-samples#12]
Hello,

TL;DR: **Please fix the SDK so that `Session.ErrorEventArgs.ErrorCode` provides a meaningful value.**

* Where? On any sample here, or any code using the SDK for that matter.
* When? On any error.
* Context? All hint we get is a `Session.Error` event called with a parameter of type `Session.ErrorEventArgs` which has a property `e.ErrorCode`.
* Expected: `e.ErrorCode` can be decoded via some documentation or enum value, like "token expired", "token unknown", "token is valid but not associated with this session", "connection closed", "yo mama pulled the cord" ;-).
* Observed: `e.ErrorCode` seems totally random, never the same.

Example values observed in identical or near-identical situations, seconds or minutes apart:

```
226401872
229336616
230507208
221168832
178993368
295706272
227620720
227614944
225505720
225962952
92129144
307975640
201365880
289036712
```

## This bug is has serious consequence.

* This has had a **high impact** today on a client demo. 
* Another bug occurred on a machine. Session error with no meaningful error code gave no hint.
* We tried to reproduce that other bug on another machine then a third machine of ours, and it seemed we could reproduce it. 
* So, we came back and have tried on our machine to get a clean log to help you reproduce.
* Alas not: when we believed we reproduced it, it was actually another error: we used a wrong token when trying to reproduce. But the SDK does not tell that, it just says "session error" and a meaningless number.
* Now we've left client premises and figured out we cannot actually reproduce the actual bug.
* So, there is a genuine error we had on a client machine but no log to send to you.
* **Had we had a meaningful error code, we would have figured out we did not have a repro, and be careful of getting the log there.**

TL;DR: **Please fix the SDK so that `e.ErrorCode` provides a meaningful value.**

**Comment by MichaelJolley** [comment]
@fidergo-stephane-gourichon, this was great feedback.  I'm starting to review our repos and found that no one responded to you.  This was resolved some time in the past and now the `ErrorEventArgs.ErrorCode` is an enum that provides more details about the error that occurred.  You can see details about the error codes at https://tokbox.com/developer/sdks/windows/reference/namespace_open_tok.html#a593c40ac342238827568a038ac099d94
